### PR TITLE
Added children to fireAfterUnDeleteEvent event

### LINKED ISSUE #5004: No $scriptProperties['children'] on event OnResourceUndelete
pureppl created Redmine issue ID 5004

OnResourceDelete has $scriptProperties['children'] but OnResourceUndelete not.

### PR DESCRIPTION
**Pull request by OptimusCrime**
Fixes #5004

### CODE REVIEW

**Comment by OptimusCrime** [comment]
Bump, this fix should be rather simple. Does not break anything, just offers extended access.

**Comment by OptimusCrime** [comment]
Anything to do with this?

File: core/model/modx/processors/resource/undelete.class.php [hunk 1]
```diff
@@ -93,6 +95,50 @@ public function removeLock() {
...
+
+    /**
+     * Helper method for getChildrenIds for getting Children recursively
+     *
+     * @see getChildrenIds
+     * @param modResource $parent
+     * @return void
+     */
+    protected function getChildren(modResource $parent) {
+        $childResources = $parent->getMany('Children');
```
Comment (TheBoxer) [inline core/model/modx/processors/resource/undelete.class.php]: I don't think this is correct, as you'll get all children by this, not only those that were un-deleted.

Comment (OptimusCrime) [inline core/model/modx/processors/resource/undelete.class.php]: Ah, right. I'll take a look at that.

**Comment by Mark-H** [comment]
Any update on making it only grab the ones that were undeleted?

**Comment by OptimusCrime** [comment]
I'll fix it

**Comment by Mark-H** [comment]
You're the awesomest

**Comment by rthrash** [comment]
This pull request has been mentioned on **MODX Community**. There might be relevant details there:

https://community.modx.com/t/discrepancy-in-event-triggering-when-deleting-undeleting-resources/4952/2

**Comment by rthrash** [comment]
This pull request has been mentioned on **MODX Community**. There might be relevant details there:

https://community.modx.com/t/discrepancy-in-event-triggering-when-deleting-undeleting-resources/4952/4

**Pull request closed without merging**